### PR TITLE
GitHub action update bot

### PR DESCRIPTION
**Pull request by reyang**
Stealing from https://github.com/open-telemetry/opentelemetry-cpp/pull/1246.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    labels:
+      - "GHA"
```
Comment (reyang) [inline .github/dependabot.yml:8]: Or probably use the `infra` tag?

Comment (cijothomas) [reply .github/dependabot.yml:8]: yes please ! we have 30+ labels already 😄 https://github.com/open-telemetry/opentelemetry-dotnet/labels

**Review by cijothomas — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (reyang) [reply .github/dependabot.yml]: Done.

**Review by cijothomas — APPROVED** [review]

**Comment by reyang** [comment]
> yes please ! we have 30+ labels already 😄 https://github.com/open-telemetry/opentelemetry-dotnet/labels

Regarding this, maybe consider clean up the labels that are not used (e.g. `wontfix` and `duplidate` seem to be useless because they should be closed immediately...)

**Comment by cijothomas** [comment]
> > yes please ! we have 30+ labels already 😄 https://github.com/open-telemetry/opentelemetry-dotnet/labels
> 
> Regarding this, maybe consider clean up the labels that are not used (e.g. `wontfix` and `duplicate` seem to be useless because they should be closed immediately...)
> 
> e.g. https://github.com/open-telemetry/opentelemetry-dotnet/issues?q=is%3Aissue+is%3Aopen+label%3Aduplicate+

did some cleanups..

**Comment by cijothomas** [comment]
@reyang Unfortunately this has issues, I am not sure if this is solved in other repos. Dependabot's PR needs to be rebased with main, and when we do it, the EasyCLA will complain.. I remember other repos reported this issue, need to see how it was resolved.

**Comment by cijothomas** [comment]
> @reyang Unfortunately this has issues, I am not sure if this is solved in other repos. Dependabot's PR needs to be rebased with main, and when we do it, the EasyCLA will complain.. I remember other repos reported this issue, need to see how it was resolved.

Only has issues in contrib repo.
image

will review the branch protection rules to see if this can be fixed.

**Comment by cijothomas** [comment]
fixed.

**Pull request merged**